src/WinRT.Runtime/WinRT.Runtime.csproj

```diff
@@ -4,7 +4,7 @@
     <TargetFrameworks>$(LibBuildTFMs)</TargetFrameworks>
     <RootNamespace>WinRT</RootNamespace>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-    <LangVersion>9</LangVersion>
+    <LangVersion>11.0</LangVersion>
```

Comment:
This is technically a breaking change for the embedded version of CsWinRT. I think we should consider bumping that but we might need a major release for that (maybe along with the rest of AOT changes?). cc. @manodasanW

Reply:
From what I understand, doesn't the embedded version use a compiled version of WinRT.Runtime, unless if we are going out our way to build our own WinRT.Runtime?

Though it might break some tests, yeah. I haven't checked.

Reply:
No it's shipping all raw source files in the NuGet package so consumers can build it locally into their assemblies. If it was a compiler version of the runtime it wouldn't really be embedded, as it'd be a separate assembly 😄

Reply:
The safest version would probably be 10.0 (.NET 6 default), since that's the oldest version that C#/WinRT officially supports. However, it lacks the UTF8 string literals feature, which is quite extensively used in this PR.

Reply:
Incrementing the lang version is a concern for the reasons mentioned.  If UTF8-string literals were why we had to increment it so that we can pass them as pointers, what is the thought on using the pinnable string marshaling from [here](https://github.com/microsoft/CsWinRT/blob/546cffe742100fdd4bf91eb29486fabf5d22d742/src/cswinrt/strings/WinRT.cs#L169-L172).

Reply:
Basically making the other string parameters to what the WindowsCreateStringReference method signature was before this change.

Reply:
We could just pass in the raw bytes of the UTF-8 string or use `Encoding.UTF8`. The first approach is faster but hurts code readability. The second approach is more readable but makes the code slower.

Reply:
Looking at the generated code of the `LibraryImport` source generator, it looks like they are using `Encoding.UTF8`.  I think we use `Encoding.UTF8` here too.  That would also work with all scenarios like `LoadLibrary(_currentModuleDirectory...)`.

Reply:
What if we could cross compile with .NET 7 and use `#if NET7_0_OR_GREATER`? Since .NET 7 default is C# 11, which allows UTF-8 string literals.

Reply:
That is a good question.  The issue is `WinRT.Runtime` is distributed in a targeting package `Microsoft.Windows.SDK.NET.Ref` which gets ingested by the .NET SDK which is the primary way folks get it and unless we start shipping a .NET 6 and .NET 7 version of the package, which I want to avoid at the moment, I don't think I can have 2 versions of the DLL and have the appropriate one included based on TFM.

This would also mean that all projections that are generated with CsWinRT would also need to generate a .NET 7 version for apps to get an AOT compatible version which I want to avoid until there is a real need to.

Reply:
> This would also mean that all projections that are generated with CsWinRT would also need to generate a .NET 7 version for apps to get an AOT compatible version which I want to avoid until there is a real need to.

I meant that we would use UTF-8 literals for .NET 7 and `Encoding.UTF8` for .NET 6 and older. Both will still benefit from the AOT-friendly Platform P/Invokes, just that the .NET 6 version will be slower due to encoding overhead.

However, of course, this still means we need to cross target .NET 6 and .NET 7, which as you said is currently undesirable. How about we would currently use `Encoding.UTF8` for both runtimes (and build a single binary for .NET 6) and then reinvestigate using UTF-8 string literals when there are more opportunities for breaking changes?

Reply:
If `Encoding.UTF8` is the fallback, then I can see doing what you suggested.  Actually, given this is a language feature, I can see also using it for `WinRT.Runtime` and the `Windows SDK projection` as we can continue to target .NET 6 but build using the .NET 7 SDK and have language version set to 11 for our builds while not breaking ones that are doing embedded.

To allow that, for the ifdefs we should do `#if NET7_0_OR_GREATER || CsWinRT_LANG_11_FEATURES` or something along those lines and then define `CsWinRT_LANG_11_FEATURES` in `WinRT.Runtime.csproj` along with `LangVersion 11`.

Reply:
Should be done in 25ab8f9ffe47352341c42fd3d29958cd5cd06899